components/config-mgmt-service/backend/elastic/nodes.go

```diff
@@ -191,23 +191,28 @@ func (es Backend) GetNodesPageByCursor(ctx context.Context, start time.Time,
 //   }
 // }
 func (es Backend) GetNodes(page int, perPage int, sortField string,
-	ascending bool, filters map[string][]string) ([]backend.Node, error) {
+	ascending bool, filters map[string][]string, startDate, endDate string) ([]backend.Node, error) {
 	// Decrement one to the page since we must start from zero
-	page = page - 1
-	start := perPage * page
+	startPage := perPage * (page - 1)
 
 	// Adding the exists = true filter to the list of filters, because nodes
 	// have documents that persist in elasticsearch to hold historical data
 	// even after the node no longer exists
 	filters["exists"] = []string{"true"}
 
-	filtersQuery := newBoolQueryFromFilters(filters)
+	mainQuery := newBoolQueryFromFilters(filters)
+
+	rangeQuery, ok := newRangeQuery(startDate, endDate, NodeCheckin)
```

Comment:
does this mean filters are ignored when doing a range query?

Reply:
No, filters are added to the mainQuery and when there are date ranges it is added to the mainQuery also. 
I will add a test to ensure that is the case.

Reply:
Here is a test for that https://github.com/chef/automate/pull/3340/files#diff-d9bd977013f859bb8b84713df8afb223R413